tutorial/hello-key/unikernel.ml

```diff
@@ -3,14 +3,14 @@ open Cmdliner
 
 let hello =
   let doc = Arg.info ~doc:"How to say hello." [ "hello" ] in
-  Arg.(value & opt string "Hello World!" doc)
+  Mirage_runtime.register_arg Arg.(value & opt string "Hello World!" doc)
```

Comment:
I really like that ! Thank you @hannesm !
I just have a suggestion (in reality more a question about why not use that idom?)
```suggestion
  Mirage_runtime.register_arg Arg.(value & opt string "Hello World!" doc) ()
```
And then leave line 13 unchanged?

Reply:
This will result in a runtime error: we need to register the arguments at initialization time, and MirageOS does magic processing, and only when `start` is called, we can use their values.

Reply:
If you call the function "too early" you will get an error /o\ See also the discussion in the PR https://github.com/mirage/mirage/pull/1574

Reply:
yes :)
```
            |      ___|
  __|  _ \  |  _ \ __ \
\__ \ (   | | (   |  ) |
____/\___/ _|\___/____/
Solo5: Bindings version v0.8.1
Solo5: Memory map: 512 MB addressable:
Solo5:   reserved @ (0x0 - 0xfffff)
Solo5:       text @ (0x100000 - 0x1c4fff)
Solo5:     rodata @ (0x1c5000 - 0x1f5fff)
Solo5:       data @ (0x1f6000 - 0x286fff)
Solo5:       heap >= 0x287000 < stack < 0x20000000
Fatal error: exception Invalid_argument("Called too early. Please delay this call to after the start function of the unikernel.")
```
I'd run it before asking /o\\

Reply:
and here I hope you get a reasonable backtrace? :)

Reply:
Instead of `unit` it could take a magic value that is passed to `start` to help guard against this error. But I'm not sure it's worth it tbh...

Reply:
My only additional line is `Solo5: solo5_exit(2) called`, not really a nice backtrace :'(

Reply:
I'll opam update+upgrade to be sure I'm not outdated

Reply:
Ouch! I guess the exception was raised before `Printexc.record_backtrace true` was called.

Reply:
meh, we should investigate. Since that's a likely error condition, and having a source location would be very beneficial.

@reynir not sure that it is worth it... But maybe the API doc should say `register_arg : .. -> (unit -> 'a)` -- introducing unnecessary parens to make it a bit more obvious to the reader that a function will be returned..

Reply:
ocamlformat will undo that straight away, unfortunately. We could write:
```OCaml
type 'arg get_arg = unit -> 'arg
val register_arg : 'a Cmdliner.Term.t -> 'a get_arg
```
But I feel we are then obfuscating...